include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -581,6 +581,9 @@ struct CustomOpApi {
   template <typename T>  // T is only implemented for float, int64_t, and string
   T KernelInfoGetAttribute(_In_ const OrtKernelInfo* info, _In_ const char* name);
 
+  template <typename T>  // T is only implemented for float, and int64_t
+  std::vector<T> KernelInfoGetAttributeArray(_In_ const OrtKernelInfo* info, _In_ const char* name);
+
   OrtTensorTypeAndShapeInfo* GetTensorTypeAndShape(_In_ const OrtValue* value);
```

Comment:
Do we need a new method or can T in KernelInfoGetAttribute be std::vector<...> with type_traits being used in the implementation to handle that case?

Reply:
We could - the current C++ approach is to explicitly specialize the template for different types - we could do the same for std::vector<...> but the C API would have different method(s) added to support parsing an attribute array whereas the C++ API would stick to one interface for primitive types and std::vector<...>. This should be fine but just calling that out...

Reply:
> > } else {
> 
> Do you know why we remove the trailing \0?
> 
> Refers to: include/onnxruntime/core/session/onnxruntime_cxx_inline.h:893 in [88e9541](https://github.com/microsoft/onnxruntime/commit/88e95415b025df4d049293b23915c0306312d8ae). [](commit_id = [88e9541](https://github.com/microsoft/onnxruntime/commit/88e95415b025df4d049293b23915c0306312d8ae), deletion_comment = False)

My understanding is that the "size" returned by the C API accounts for the trailing null character and when we actually call api_.KernelInfoGetAttribute_string(...), the underlying character array now contains the trailing null character too which isn't needed because anyone using `c_str()` on the std::string object will get a null terminated c string anyway (even if the underlying char array doesn't contain a null character).